spec/unit/indirector/catalog/compiler_spec.rb

```diff
@@ -414,6 +426,25 @@ def a_request_that_contains(facts)
     end
   end
 
+  describe "in an IPv6 only environment" do
+    before do
+      allow(Puppet).to receive(:version).and_return(1)
+      set_facts({
+        'fqdn'       => "my.server.com",
+        'ipaddress'  => nil,
+        'ipaddress6' => "my.ipv6.address"
+        })
+      @request = Puppet::Indirector::Request.new(:catalog, :find, node_name, nil)
+      allow(compiler).to receive(:compile)
+      allow(Puppet::Node.indirection).to receive(:find).with(node_name, anything).and_return(node)
+    end
+
+    it "should use the IPv6 address for serverip if there is no IPv4 address" do
```

Comment:
minor nit, we decided not to populate `serverip` with the IPv6, so this description needs updating